### PR TITLE
Flowsheet documentation for NaCl crystallizer

### PR DESCRIPTION
**Pull request by OOAmusat**
## Fixes/Resolves:
NaCl crystallizer flowsheet documentation for #1219

## Summary/Motivation:
Adding NaCl crystallizer flowsheet documentation

## Changes proposed in this PR:
- Only includes adding .rst for the NaCl crystallizer example

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: docs/technical_reference/flowsheets/crystallization.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
+NaCl Crystallization
+===============================
+
+Introduction
+------------
+
+Crystallization is the precipitation and extraction of the crystals from a mother liquor based on solute supersaturation. Crystallization is a potential solution to desalination's environmental brine management problem; brine crystallization systems shown to achieve zero-liquid discharge and salt recovery.
```
Comment (MarcusHolly) [inline docs/technical_reference/flowsheets/crystallization.rst]:
```suggestion
Crystallization is the precipitation and extraction of the crystals from a mother liquor based on solute supersaturation. Crystallization is a potential solution to desalination's environmental brine management problem; brine crystallization systems have been shown to achieve zero-liquid discharge and salt recovery.
```

File: docs/technical_reference/flowsheets/crystallization.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+The objective of the mathematical model is to simulate crystallization from a NaCl mother liquor. The example is based on (but does not replicate exactly) Example 11.4 in `Tavare, N.S. (1995). <https://link.springer.com/chapter/10.1007/978-1-4899-0233-7_11>`_ 
+
+Implementation
+--------------
+
+The modeled crystallization proecess is illustrated by Figure 1. Thermal energy for the evaporative crystallizer is added in the suspension heater, with the water vapor product sent off to a condenser (not modeled here). 
+
```
Comment (MarcusHolly) [inline docs/technical_reference/flowsheets/crystallization.rst]:
```suggestion
The modeled crystallization process is illustrated by Figure 1. Thermal energy for the evaporative crystallizer is added in the suspension heater, with the water vapor product sent off to a condenser (not modeled here). 

```

File: docs/technical_reference/flowsheets/crystallization.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+   * a property package supporting all three phases (e.g., the crystallizer property package) is provided for the crystallizer model.
+
+.. figure:: ../../_static/flowsheets/crystallizer.jpg
+    :width: 500
+    :align: center
+
+    Figure 1. Crystallizer flowsheet
+
+Documentation for the crystallizer unit model can be found below. 
+    * `Crystallizer <https://watertap.readthedocs.io/en/latest/technical_reference/unit_models/crystallizer_0D.html>`_
```
Comment (MarcusHolly) [inline docs/technical_reference/flowsheets/crystallization.rst:28]: Consider replacing the image in the crystallizer unit model documentation with the nicer image you've added above (I'm assuming they're depicting the same process).

File: docs/technical_reference/flowsheets/crystallization.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+Degrees of Freedom
+------------------
+For the crystallizer unit model, if the inlet feed condition is fully specified, the user is left with two degrees of freedom. In this example, the following variables are initially specified for simulating the crystallizer (i.e., degrees of freedom = 0):
+
+    * NaCl mother liqour conditions (i.e., flow, temperature, pressure, compositions)
```
Comment (MarcusHolly) [inline docs/technical_reference/flowsheets/crystallization.rst]:
```suggestion
    * NaCl mother liquor conditions (i.e., flow, temperature, pressure, compositions)
```

File: docs/technical_reference/flowsheets/crystallization.rst [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    * crystallizer operating temperature
```
Comment (MarcusHolly) [inline docs/technical_reference/flowsheets/crystallization.rst]:
```suggestion
    * Crystallizer operating temperature
```
Just to be consistent with other capitalization in this section

File: docs/technical_reference/flowsheets/crystallization.rst [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    * Magma density
+    * Vapour outlet mass flow
+    * Crystallizer thermal energy input
+
+We demonstrate how this may be done in our flowsheet by updating and re-solving the model. Thus, the example shows five cases:
+#. **Case 1**: fixing the crystallizer operating temperature and solid outlet mass flow
+#. **Case 2**: fixing the crystallizer operating temperature and crystallization yield
+#. **Case 3**: fixing the crystallizer operating temperature and the product slurry solids volumetric fraction
+#. **Case 4**: fixing the crystallizer operating temperature and magma density
+#. **Case 5**: fixing the crystallizer operating temperature and crystallizer thermal energy input
```
Comment (MarcusHolly) [inline docs/technical_reference/flowsheets/crystallization.rst]: I'm assuming you wanted to display these as bullets, but they're just coming across as a regular paragraph in the documentation. So I think you would need to match the bulleting format you've used above or add in some blank lines.

**Review by MarcusHolly — COMMENTED** [review]
A few minor comments to address

File: docs/technical_reference/flowsheets/crystallization.rst (hunk 1 above)
Comment (OOAmusat) [reply docs/technical_reference/flowsheets/crystallization.rst]: Done.

File: docs/technical_reference/flowsheets/crystallization.rst (hunk 3 above)
Comment (OOAmusat) [reply docs/technical_reference/flowsheets/crystallization.rst:28]: Done.

File: docs/technical_reference/flowsheets/crystallization.rst (hunk 4 above)
Comment (OOAmusat) [reply docs/technical_reference/flowsheets/crystallization.rst]: Done.

File: docs/technical_reference/flowsheets/crystallization.rst (hunk 6 above)
Comment (OOAmusat) [reply docs/technical_reference/flowsheets/crystallization.rst]: Done.

File: docs/technical_reference/flowsheets/crystallization.rst (hunk 5 above)
Comment (OOAmusat) [reply docs/technical_reference/flowsheets/crystallization.rst]: Done.

File: docs/technical_reference/flowsheets/crystallization.rst (hunk 2 above)
Comment (OOAmusat) [reply docs/technical_reference/flowsheets/crystallization.rst]: Done.

**Review by MarcusHolly — APPROVED** [review]
LGTM

File: docs/technical_reference/flowsheets/crystallization.rst [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+   * a property package supporting all three phases (e.g., the crystallizer property package) is provided for the crystallizer model.
+
+.. figure:: ../../_static/flowsheets/crystallizer.jpg
+    :width: 500
+    :align: center
+
+    Figure 1. Crystallizer flowsheet
+
+Documentation for the crystallizer unit model can be found below. 
+    * `Crystallizer <https://watertap.readthedocs.io/en/latest/technical_reference/unit_models/crystallizer_0D.html>`_
```
Comment (MarcusHolly) [inline docs/technical_reference/flowsheets/crystallization.rst:28]: One more thing I missed in my review - you should also provide a link here to the crystallization property package. Although it looks like there's no documentation for this yet... I think it'd be nice to have this for the 1.0 release, so maybe it's best to address this in a separate PR that also adds the prop pack documentation.

**Review by agarciadiego — APPROVED** [review]

**Pull request merged**